src/plugins/dashboard/public/dashboard_app/dashboard_app.test.tsx

```diff
@@ -0,0 +1,105 @@
+/* @link https://github.com/elastic/kibana/pull/190086/
+*/
+
+describe('Dashboard App', () => {
+  const mockDashboard = buildMockDashboard();
+  const expandedPanelIdSpy = jest.spyOn(mockDashboard.expandedPanelId, 'next');
+  let mockHistory: MemoryHistory;
+  let historySpy: jest.SpyInstance;
+  beforeAll(() => {
+    mockHistory = createMemoryHistory();
+    historySpy = jest.spyOn(mockHistory, 'replace');
+    /**
+     * Mock the LazyDashboardRenderer component to avoid rendering the actual dashboard
+     * and hitting errors that aren't relevant
+     */
+    jest
+      .spyOn(dashboardRendererStuff, 'LazyDashboardRenderer')
+      // we need overwrite the onApiAvailable prop to get the dashboard Api in the dashboard app
+      .mockImplementation(({ onApiAvailable }: DashboardRendererProps) => {
+        useEffect(() => {
+          onApiAvailable?.(mockDashboard as DashboardApi);
+        }, [onApiAvailable]);
+
+        return <div>Test renderer</div>;
+      });
+  });
+
+  beforeEach(() => {
+    // reset the mock history calls per test
+    historySpy.mockClear();
+    // reset the expanded panel id between tests
+    expandedPanelIdSpy.mockClear();
+  });
+
+  it('test the default behavior without an expandedPanel id passed as a prop to the DashboardApp', async () => {
+    // default conditions with no expanded panel id and no urls being passed with expanded panel ID in the url
+    render(<DashboardApp redirectTo={jest.fn()} history={mockHistory} />);
+
+    await waitFor(() => {
+      // this value should be undefined by default
+      expect(mockDashboard.expandedPanelId.getValue()).toBe(undefined);
+      // history should not be called
+      expect(historySpy).toHaveBeenCalledTimes(0);
+      expect(mockHistory.location.pathname).toBe('/');
+    });
+
+    // simulate expanding a panel
+    mockDashboard.expandPanel('123');
+
+    await waitFor(() => {
+      expect(mockDashboard.expandedPanelId.getValue()).toBe('123');
+      expect(historySpy).toHaveBeenCalledTimes(1);
+      expect(mockHistory.location.pathname).toBe('/create/123');
+    });
+  });
+
+  it('test that the expanded panel behavior subject and history is called when passed as a prop to the DashboardApp', async () => {
+    // this tests a user getting a url from another user that has an expanded panel on the dashboard
+    // test someone giving a url with an expanded panel id, as if it goes through the router
+    mockHistory.push('/create/456');
+
+    render(
+      <DashboardApp
+        savedDashboardId=""
+        redirectTo={jest.fn()}
+        history={mockHistory}
+        expandedPanelId="456"
+      />
+    );
+
+    await waitFor(() => {
+      expect(expandedPanelIdSpy).toHaveBeenCalledTimes(1);
+      // history replace should not be called until minimizing the panel
+      expect(historySpy).toHaveBeenCalledTimes(0);
+      expect(mockDashboard.getExpandedPanelId());
+      expect(mockHistory.location.pathname).toBe('/create/456');
```

Comment:
I don't think we should test `pathname` on load - and we should remove the `mockHistory.push('/create/456');` above :) This seems circular - the `mockHistory.push('/create/456');` is what is making this `expect` true, so we aren't actually testing anything 🤷

Reply:
Yeah Im having issues figuring out how to test this part. So I want to capture as if a user is given a url to an expanded panel. In that case, the dashboard router passes the expand panel Id to the dashboard app which then passes the expandedPanel to the dashboardRenderer...

Reply:
I'm attempting to show that a url with the expanded panel ID in it would go through the mountApp() in the DashboardRouter in [d1097d0](https://github.com/elastic/kibana/pull/190086/commits/d1097d07de54e6eb859f3f38c21c02e756cc35e3) but it isn't showing in the DashboardApp with `expect(mockDashboard.expandedPanelId.getValue()).toBe('456');` but is instead showing that value as undefined...I think the issue might be that the router matches the expandedPanelId and then passes that to the DashboardApp which then is passing it to the renderer. I'm thinking it might be too many moving parts for a unit test and that's why the subscription isn't working in the test? WDYT?